beacon-chain/state/state-native/v1/getters_eth1.go

```diff
@@ -6,86 +6,66 @@ import (
 
 // Eth1Data corresponding to the proof-of-work chain information stored in the beacon state.
 func (b *BeaconState) Eth1Data() *ethpb.Eth1Data {
-	if !b.hasInnerState() {
-		return nil
-	}
-	if b.state.Eth1Data == nil {
+	if b.eth1Data == nil {
 		return nil
 	}
 
 	b.lock.RLock()
 	defer b.lock.RUnlock()
 
-	return b.eth1Data()
+	return b.eth1DataInternal()
 }
 
-// eth1Data corresponding to the proof-of-work chain information stored in the beacon state.
+// eth1DataInternal corresponding to the proof-of-work chain information stored in the beacon state.
 // This assumes that a lock is already held on BeaconState.
-func (b *BeaconState) eth1Data() *ethpb.Eth1Data {
-	if !b.hasInnerState() {
-		return nil
-	}
-	if b.state.Eth1Data == nil {
+func (b *BeaconState) eth1DataInternal() *ethpb.Eth1Data {
+	if b.eth1Data == nil {
 		return nil
 	}
 
-	return ethpb.CopyETH1Data(b.state.Eth1Data)
+	return ethpb.CopyETH1Data(b.eth1Data)
 }
 
 // Eth1DataVotes corresponds to votes from Ethereum on the canonical proof-of-work chain
 // data retrieved from eth1.
 func (b *BeaconState) Eth1DataVotes() []*ethpb.Eth1Data {
-	if !b.hasInnerState() {
-		return nil
-	}
-	if b.state.Eth1DataVotes == nil {
+	if b.eth1DataVotes == nil {
 		return nil
 	}
 
 	b.lock.RLock()
 	defer b.lock.RUnlock()
 
-	return b.eth1DataVotes()
+	return b.eth1DataVotesInternal()
 }
 
-// eth1DataVotes corresponds to votes from Ethereum on the canonical proof-of-work chain
+// eth1DataVotesInternal corresponds to votes from Ethereum on the canonical proof-of-work chain
 // data retrieved from eth1.
 // This assumes that a lock is already held on BeaconState.
-func (b *BeaconState) eth1DataVotes() []*ethpb.Eth1Data {
-	if !b.hasInnerState() {
-		return nil
-	}
-	if b.state.Eth1DataVotes == nil {
+func (b *BeaconState) eth1DataVotesInternal() []*ethpb.Eth1Data {
+	if b.eth1DataVotes == nil {
 		return nil
 	}
 
-	res := make([]*ethpb.Eth1Data, len(b.state.Eth1DataVotes))
+	res := make([]*ethpb.Eth1Data, len(b.eth1DataVotes))
 	for i := 0; i < len(res); i++ {
-		res[i] = ethpb.CopyETH1Data(b.state.Eth1DataVotes[i])
+		res[i] = ethpb.CopyETH1Data(b.eth1DataVotes[i])
 	}
 	return res
 }
 
 // Eth1DepositIndex corresponds to the index of the deposit made to the
 // validator deposit contract at the time of this state's eth1 data.
 func (b *BeaconState) Eth1DepositIndex() uint64 {
-	if !b.hasInnerState() {
-		return 0
-	}
-
 	b.lock.RLock()
 	defer b.lock.RUnlock()
 
-	return b.eth1DepositIndex()
+	return b.eth1DepositIndexInternal()
 }
 
 // eth1DepositIndex corresponds to the index of the deposit made to the
 // validator deposit contract at the time of this state's eth1 data.
 // This assumes that a lock is already held on BeaconState.
-func (b *BeaconState) eth1DepositIndex() uint64 {
-	if !b.hasInnerState() {
-		return 0
-	}
-
-	return b.state.Eth1DepositIndex
+func (b *BeaconState) eth1DepositIndexInternal() uint64 {
```

Comment:
Why do we need another internal function if there's no additional logic other than return the underlying object? Same with genesis time, slot... etc

Reply:
Good point, I will remove these.